todo.js

```diff
@@ -1,17 +1,54 @@
 const inputElement = document.getElementById("todoInput");
-const tasksList = document.getElementById("tasksList"); //ul
+const tasksList = document.getElementById("tasksList");
 const addBtn = document.getElementById("add-btn");
 
+let todoItems = [];
+
 function handleTodo() {
   const taskText = inputElement.value;
-  const newTodo = document.createElement("li");
-  newTodo.innerHTML = taskText;
-  tasksList.appendChild(newTodo);
+
+  const newTodo = {
+    id: Date.now(),
+    value: taskText,
+    doneState: false,
+  };
+
+  todoItems.push(newTodo);
+  renderTodoList();
   resetInputField();
 }
 
 function resetInputField() {
   inputElement.value = "";
 }
 
+function renderTodoList() {
+  tasksList.innerHTML = "";
+
+  todoItems.forEach((todo) => {
+    const newTodoItem = document.createElement("li");
+    newTodoItem.innerHTML = todo.value;
+
+    const deleteButtonElement = createDeleteButton(todo.id);
+    newTodoItem.appendChild(deleteButtonElement);
+    tasksList.appendChild(newTodoItem);
+  });
+}
+
+function createDeleteButton(todoId) {
+  const deleteButtonElement = document.createElement("button");
+  deleteButtonElement.innerHTML = "Delete";
+  deleteButtonElement.addEventListener("click", function () {
+    deleteTodoItem(todoId);
+  });
+  return deleteButtonElement;
+}
+
+function deleteTodoItem(todoId) {
+  todoItems = todoItems.filter((todo) => todo.id !== todoId);
+  renderTodoList();
+}
```

Comment:
Let's follow single responsibility principle in this function.
This function as it's name suggests is supposed to delete a todo item. Instead it's also calling a render() function. 

Let's move `renderTodoList()` call where we are calling `deleteTodoItem()` method.

The flow will be:
1. User clicks on delete button of a specific todo
2. The handler method is called for that buttons
3. Handler method calls `deleteTodoItem` with the id
4. After the above call we render the list again.